webapp/src/shared/components/templates/RowsPageTemplate/RowsPageTemplate.tsx

```diff
@@ -0,0 +1,82 @@
+import './RowsPageTemplate.css';
+import * as React from 'react';
+import { Link } from 'react-router-dom';
+import { USER_URL, AVATAR_EP_URL, WILDCARD_AVATAR_URL } from '../../../urls';
+import { MediumAvatar } from '../../Avatar/Avatar';
+import SearchForm from '../../Input/SearchForm';
+import RowsList, { RowItem } from '../../RowsList/RowsList';
+import NavigationBar from '../../NavigationBar/NavigationBar';
+import { useData } from '../../../hooks/UseData';
+import Loading from '../../Loading/Loading';
+import { usersApi } from '../../../services/ApiService';
+import { useSearchContext } from '../../../context/SearchContext';
+
+type RowsPageTemplateProps<T> = {
+  dataValidator: (data: T) => boolean;
+  dataMapper: (data: T) => RowItem;
+};
+
+export default function RowsPageTemplate<T>({
+  dataValidator,
+  dataMapper,
+}: RowsPageTemplateProps<T>) {
+  const { result, fetchMoreResults } = useSearchContext();
+
+  const data = ((array: object): RowItem[] | null => {
+    let rows = [];
+
+    if (!Array.isArray(array)) {
+      return null;
+    }
+    for (let i = 0; i < array.length; i++) {
+      if (!dataValidator(array[i])) {
```

Comment:
agree, is sadly less readable to use a for loop than a map but, since we are sanitizing arrays with no limits in length, I think we gain a bit in performance saving an iteration. A web application could run on the latest iphone or on a 8 years old phone, depending on the device that could be expensive

Reply:
I have to say that we added the type validator before we had properly generated Api files. We didn't have proper type checking from the request methods. I think now the generated fetching methods provide runtime type checking. Maybe it's obsolete, and we don't really need this here at all anymore.

Reply:
Agree, data validation is not needed anymore, removed and simplified a bit this method!